devices/Bmm150/README.md

```diff
@@ -0,0 +1,181 @@
+
+I2cConnectionSettings mpui2CConnectionSettingmpus = new(1, Bmm150.DefaultI2cAddress);
+
+using Bmm150 bmm150 = new Bmm150(I2cDevice.Create(mpui2CConnectionSettingmpus));
+
+Debug.WriteLine($"Please move your device in all directions...");
+
+bmm150.CalibrateMagnetometer();
+
+Debug.WriteLine($"Calibration completed.");
+
+while (true)
+{
+    Vector3 magne = bmm150.ReadMagnetometer(true, TimeSpan.FromMilliseconds(11));
+
+    Debug.WriteLine($"Mag data: X={magne.X,15}, Y={magne.Y,15}, Z={magne.Z,15}");
+
+    Thread.Sleep(100);
+}
+```
+### Expected output
+
+```console
+Please move your device in all directions...
+Calibration completed.
+Mag data: X=    19.79405975, Y=   -38.48844909, Z=   -17.28075408
+Mag data: X=     9.16391563, Y=   -25.65896606, Z=   -26.12207031
+Mag data: X=     5.49845027, Y=    -4.39876031, Z=   -28.53705596
+Mag data: X=     8.79735946, Y=    -3.66556644, Z=   -17.68263244
+Mag data: X=    12.82971763, Y=    -4.39876031, Z=   -19.29265785
+Mag data: X=    13.92915248, Y=     2.93245315, Z=    -5.22441434
+Mag data: X=    14.66280078, Y=    -0.36657002, Z=    -2.81387901
+Mag data: X=    12.46315383, Y=    1.099690079, Z=     5.62702512
+Mag data: X=    10.63053035, Y=     3.66570019, Z=     6.43172311
+Mag data: X=     6.23157691, Y=     0.36656334, Z=     2.81351256
+Mag data: X=     3.66549944, Y=              0, Z=      2.8127799
+Mag data: X=     2.56589651, Y=              0, Z=     4.42065811
+Mag data: X=     0.73312669, Y=    -2.19938015, Z=    6.028955459
+```
+
+## Calibration
+
+You can get access perfom calibration thru the ```CalibrateMagnetometer``` function which will. Be aware that the calibration takes a few seconds.
+
+```csharp
+bmm150.CalibrateMagnetometer();
+```
+
+If no calibration is performed, you will get a raw data cloud which looks like this:
+
+![raw data](./rawcalib.png)
+
+Running the calibration properly require to **move the sensor in all the possible directions** while performing the calibration. You should consider running it with enough samples, at least few hundreds. The default is set to 1000. While moving the sensor in all direction, far from any magnetic field, you will get the previous clouds. Calculating the average from those clouds and subtracting it from the read value will give you a centered cloud of data like this:
+
+![raw data](./corrcalib.png)
+
+To create those cloud point graphs, every cloud is a coordinate of X-Y, Y-Z and Z-X. 
+
+Once the calibration is done, you will be able to read the data with the bias corrected using the ```ReadMagnetometer``` function. You will still be able to read the data without any calibration using the ```ReadMagnetometerWithoutCalibration``` function.
+
+## Using a different I2C interface
+
+This sensor is used for example in the [MPU9250](../Mpu9250/README.md). The MPU9250 is in this case a master I2C controlling the slave Bmm150 I2C sensor. An abstract class is available to implement basic I2C operation:
```

Comment:
this is not relevant here. See comments in the code. Check how the MBI160 is embedding this sensor.

Reply:
solved